src/main/java/com/jcabi/log/MulticolorLayout.java

```diff
@@ -183,61 +151,46 @@ public void setColors(final String cols) {
      * @param lev JavaScript like map of levels
      * @since 0.9
      */
+    @SuppressWarnings("PMD.UseConcurrentHashMap")
     public void setLevels(final String lev) {
-        for (final String item : lev.split(MulticolorLayout.SPLIT_ITEMS)) {
-            final String[] values = item.split(MulticolorLayout.SPLIT_VALUES);
-            final String level = values[0].toUpperCase(Locale.ENGLISH);
-            if (Level.toLevel(level, null) == null) {
-                throw new IllegalArgumentException(
-                    String.format(Locale.ENGLISH, "Unknown level '%s'", level)
-                );
-            }
-            this.levels.put(level, values[1]);
+        final Map<String, String> parsed = new ParseLevelInformation(
+            lev
+        ).parse();
+        for (final Entry<String, String> entry : parsed.entrySet()) {
+            this.levels.put(entry.getKey(), entry.getValue());
         }
     }
 
     @Override
     public String format(final LoggingEvent event) {
-        final String answer;
+        final Formatted formatted;
         if (this.isColoringEnabled()) {
-            answer = this.colorfulFormatting(event);
+            formatted = this.colorful(event);
         } else {
-            answer = this.dullFormatting(event);
+            formatted = this.dull(event);
         }
-        return answer;
+        return formatted.format();
     }
 
     /**
-     * Formats a log event without using ANSI color codes.
-     * @param event Log event
-     * @return Text of a log event, not colored with ANSI color codes even
-     *  if there is markup that tells to color it.
+     * Generate a dull {@link Formatted}.
+     *
+     * @param event Event to be formatted.
```

Comment:
@jrdalpra we avoid periods at the end of at-clauses, see https://github.com/teamed/qulice/issues/733